opentelemetry-exporter-geneva/geneva-uploader/src/payload_encoder/otlp_encoder.rs

```diff
@@ -0,0 +1,410 @@
+        fields.push((
+            Cow::Borrowed(FIELD_TIMESTAMP),
+            BondDataType::BT_STRING as u8,
+        ));
+        fields.push((Cow::Borrowed(FIELD_ENV_TIME), BondDataType::BT_STRING as u8));
+
+        // Part A extension - Conditional fields
+        if !log.trace_id.is_empty() {
+            fields.push((Cow::Borrowed(FIELD_TRACE_ID), BondDataType::BT_STRING as u8));
+        }
+        if !log.span_id.is_empty() {
+            fields.push((Cow::Borrowed(FIELD_SPAN_ID), BondDataType::BT_STRING as u8));
+        }
+        if log.flags != 0 {
+            fields.push((
+                Cow::Borrowed(FIELD_TRACE_FLAGS),
+                BondDataType::BT_INT32 as u8,
+            ));
+        }
+
+        // Part B - Core log fields
+        if !log.event_name.is_empty() {
+            fields.push((Cow::Borrowed(FIELD_NAME), BondDataType::BT_STRING as u8));
+        }
+        fields.push((
+            Cow::Borrowed(FIELD_SEVERITY_NUMBER),
+            BondDataType::BT_INT32 as u8,
+        ));
+        if !log.severity_text.is_empty() {
+            fields.push((
+                Cow::Borrowed(FIELD_SEVERITY_TEXT),
+                BondDataType::BT_STRING as u8,
+            ));
+        }
+        if let Some(body) = &log.body {
+            if let Some(Value::StringValue(_)) = &body.value {
+                // Only included in schema when body is a string value
+                fields.push((Cow::Borrowed(FIELD_BODY), BondDataType::BT_STRING as u8));
+            }
+            //TODO - handle other body types
+        }
+
+        // Part C - Dynamic attributes
+        for attr in &log.attributes {
+            if let Some(val) = attr.value.as_ref().and_then(|v| v.value.as_ref()) {
+                let type_id = match val {
+                    Value::StringValue(_) => BondDataType::BT_STRING as u8,
+                    Value::IntValue(_) => BondDataType::BT_INT32 as u8,
+                    Value::DoubleValue(_) => BondDataType::BT_FLOAT as u8, // TODO - using float for now
+                    Value::BoolValue(_) => BondDataType::BT_INT32 as u8, // representing bool as int
+                    _ => continue,
+                };
+                fields.push((Cow::Owned(attr.key.clone()), type_id));
+            }
+        }
+        fields.sort_by(|a, b| a.0.cmp(&b.0)); // Sort fields by name consistent schema ID generation
+        fields
+            .into_iter()
+            .enumerate()
+            .map(|(i, (name, type_id))| FieldDef {
+                name,
+                type_id,
+                field_id: (i + 1) as u16,
+            })
+            .collect()
+    }
+
+    /// Calculate schema ID from field specifications
+    fn calculate_schema_id(fields: &[FieldDef]) -> u64 {
+        use std::collections::hash_map::DefaultHasher;
+        use std::hash::{Hash, Hasher};
+
+        let mut hasher = DefaultHasher::new();
+
+        for field in fields {
```

Comment:
Similar to https://github.com/open-telemetry/opentelemetry-rust-contrib/pull/261#discussion_r2155641517, the hash computation could also be done in the same traversal required for `determine_fields` method.

Reply:
Good point #311